lib/sqlalchemy/dialects/postgresql/base.py

```diff
@@ -2827,7 +2827,14 @@ def get_columns(self, connection, table_name, schema=None, **kw):
         table_oid = self.get_table_oid(
             connection, table_name, schema, info_cache=kw.get("info_cache")
         )
-        SQL_COLS = """
+
+        generated = (
+            "a.attgenerated as generated"
+            if self.server_version_info >= (12,)
+            else "'' as generated"
```

Comment:
probably have this "NULL AS generated"

Reply:
it returns an empty string when a column is not generated. So I made it generate an empty string for consistency. It can be changed, changing also the check below

Reply:
yeah I like NULL when the DB doesn't support it at all.  this is all nitpicking I know.